h/nipsa/worker.py

```diff
@@ -31,40 +34,26 @@ def remove_nipsa_action(index, annotation):
     }
 
 
-def add_or_remove_nipsa(client, index, userid, action):
-    """Add/remove the NIPSA flag to/from all of the user's annotations."""
-    assert action in ("add_nipsa", "remove_nipsa")
-    if action == "add_nipsa":
-        query = search.not_nipsad_annotations(userid)
-        action_func = add_nipsa_action
-    else:
-        query = search.nipsad_annotations(userid)
-        action_func = remove_nipsa_action
-
-    annotations = helpers.scan(client=client, query=query)
-    actions = [action_func(index, a) for a in annotations]
-    helpers.bulk(client=client, actions=actions)
-
-
-def worker(request):
-    """Worker function for NIPSA'ing and un-NIPSA'ing users.
+def bulk_update_annotations(client, query, action):
+    """Update annotations for a user with the given NIPSA state."""
```

Comment:
Would be nice to have a bit more documentation here, especially in regards to the `action` parameter taking a function which should return an ElasticSearch action. I can see this becoming confusing after not looking at this code for a few months.